dvc/api.py

```
@@ -69,3 +75,108 @@ def _make_repo(repo_url, rev=None):
     else:
         with external_repo(url=repo_url, rev=rev) as repo:
             yield repo
+
+
+def summon(name, fname="dvcsummon.yaml", args=None, repo=None, rev=None):
+    # TODO: Write a meaningful docstring about `summon`
+    with _make_repo(repo, rev=rev) as _repo:
+
+        def pull_dependencies(deps):
+            if not deps:
+                return
+
+            outs = [_repo.find_out_by_relpath(dep) for dep in deps]
+
+            with _repo.state:
+                for out in outs:
+                    _repo.cloud.pull(out.get_used_cache())
+                    out.checkout()
+
+        path = os.path.join(_repo.root_dir, fname)
+        obj = _get_object_from_summoners_file(name, path)
+        info = obj["summon"]
+
+        pull_dependencies(info.get("deps"))
+
+        _args = copy.deepcopy(info.get("args", {}))
+        _args.update(args or {})
+
+        return _invoke_method(info["call"], _args, path=_repo.root_dir)
+
+
+def _get_object_from_summoners_file(name, path):
+    """
+    Given a summonable object's name, search for it on the given file
+    and bring it to life.
+    """
+    schema = Schema(
+        [
+            {
+                "name": str,
+                "description": str,
+                "paper": str,
+                "metrics": dict,
+                "summon": {
+                    "type": "python",
+                    "call": str,
+                    "args": dict,  # XXX: Optional
+                    "deps": [str],  # XXX: Optional
+                },
+            }
+        ]
+    )
+
+    with open(path, "r") as fobj:
+        try:
+            objects = ruamel.yaml.safe_load(fobj.read())["objects"]
+            objects = schema(objects)
+            return next(x for x in objects if x["name"] == name)
```

Again, need to check there is only one object named `name`.

---

Do we need to raise an error? Isn't it better to just pick the first one

---

Yes we need to raise an error, otherwise it might be a nasty surprise for a user.